### PR TITLE
Update from-using-pivot-and-unpivot.md to add UNPIVOT syntax

### PR DESCRIPTION
**Pull request by Rob-S**
I was looking for the UNPIVOT syntax, but the page only has PIVOT syntax. The most significant differences for UNPIVOT include no &lt;aggregate function&gt;, no &lt;alias for the pivot table&gt; and the pivot columns from the IN clause are not repeated in the SELECT clause. Also, there were "[" and "]" rather than "&lt;" and "&gt;" identifying placeholders to be replaced by column names.

### CODE REVIEW

**Comment by Court72** [comment]
@VanMSFT

Can you review the proposed changes? 
	
**Important**: When the changes are ready for publication, adding a `#sign-off` comment is the best way to signal that the PR is ready for the review team to merge. 
	
#label:"aq-pr-triaged"
@MicrosoftDocs/public-repo-pr-review-team

File: docs/t-sql/queries/from-using-pivot-and-unpivot.md [hunk 1]
```diff
@@ -38,22 +38,43 @@ The syntax for `PIVOT` is simpler and more readable than the syntax that may oth
...
-    [first pivoted column] AS <column name>,  
-    [second pivoted column] AS <column name>,  
-    ...  
-    [last pivoted column] AS <column name>  
+SELECT [<non-pivoted column> AS <column name>,]
+    ...
+    <first pivoted column> AS <column name>,
+    <second pivoted column> AS <column name>,
+    ...
+    <last pivoted column> AS <column name>
```
Comment (VanMSFT) [inline docs/t-sql/queries/from-using-pivot-and-unpivot.md]: The original syntax here is correct. The first column is not optional and hence encased in `< >` to denote the non-pivoted column. The rest of the columns are optional, and should be encased by brackets `[ ]`

File: docs/t-sql/queries/from-using-pivot-and-unpivot.md [hunk 2]
```diff
@@ -38,22 +38,43 @@ The syntax for `PIVOT` is simpler and more readable than the syntax that may oth
...
+    (<SELECT query that produces the data>)   
+    AS <alias for the source query>  
+UNPIVOT
+(  
+    <new output column created for values in result of the source query>
+FOR   
+<output column for names of the pivot columns>
+    IN ( [first pivoted column], [second pivoted column],  
+    ... [last pivoted column])  
+)
```
Comment (VanMSFT) [inline docs/t-sql/queries/from-using-pivot-and-unpivot.md]: This syntax probably better suited in the [UNPIVOT Example](https://learn.microsoft.com/en-us/sql/t-sql/queries/from-using-pivot-and-unpivot?view=sql-server-ver16#unpivot-example) section.

**Review by VanMSFT — COMMENTED** [review]
Thanks, @Rob-S! Please see my comments and let me know your thoughts.

File: docs/t-sql/queries/from-using-pivot-and-unpivot.md (hunk 1 above)
Comment (Rob-S) [reply docs/t-sql/queries/from-using-pivot-and-unpivot.md]: > The original syntax here is correct. The first column is not optional and hence encased in < > to denote the non-pivoted column.

First, the first, non-pivoted column is optional. I can remove that from the first PIVOT example and when I run it, it works fine:

```-- Pivot table with one row and five columns  
SELECT [0], [1], [2], [3], [4]  
FROM  
(
  SELECT DaysToManufacture, StandardCost   
  FROM Production.Product
) AS SourceTable  
PIVOT  
(  
  AVG(StandardCost)  
  FOR DaysToManufacture IN ([0], [1], [2], [3], [4])  
) AS PivotTable;
```

So I believe using the "< >" to denote the column and "[ ]" to denote the whole thing, including "AS" and the alias, is correct. Technically, the AS clause is also optional, so it possibly should be:

`SELECT [<non-pivoted column> [AS <column name>], ]`

> The rest of the columns are optional, and should be encased by brackets [ ]

If that's the case, then the whole line, including "AS" and the alias should be encased in the square brackets "[ ]". The characters around the column name, ex: currently "[...]" as in "[second pivoted column]", denoting the column only, should be "&lt;...&gt;", as in "&lt;second pivoted column&gt;". Therefore, the syntax for that line would be:

`[<second pivoted column> [AS <column name>], ]`

The other lines under the SELECT clause would be similar.

File: docs/t-sql/queries/from-using-pivot-and-unpivot.md (hunk 2 above)
Comment (Rob-S) [reply docs/t-sql/queries/from-using-pivot-and-unpivot.md]: The Remarks directly below the Syntax section directly reference the UNPIVOT clause. So if the Remarks are there, then I would say the UNPIVOT should remain above those Remarks.

The other option is:
- Change "Syntax" to "PIVOT Syntax"
- Add a "UNPIVOT Syntax" Section above the "UNPIVOT Example"
- Move the first paragraph of the Remarks, concerning the UNPIVOT syntax, below the "UNPIVOT Syntax" section.

File: docs/t-sql/queries/from-using-pivot-and-unpivot.md (hunk 1 above)
Comment (VanMSFT) [reply docs/t-sql/queries/from-using-pivot-and-unpivot.md]: Hi @Rob-S - Thanks! You're right, you can run the script without a non-pivoted column. But I don't see that as very useful if that's the case. I agree with your changes though.

File: docs/t-sql/queries/from-using-pivot-and-unpivot.md (hunk 2 above)
Comment (VanMSFT) [reply docs/t-sql/queries/from-using-pivot-and-unpivot.md]: I think it's probably best if we add some more context before the syntax to include `UNPIVOT`. Thanks for the suggestions!

File: docs/t-sql/queries/from-using-pivot-and-unpivot.md [hunk 3]
```diff
@@ -38,23 +38,43 @@ The syntax for `PIVOT` is simpler and more readable than the syntax that may oth
 The following syntax summarizes how to use the `PIVOT` operator.
```
Comment (VanMSFT) [inline docs/t-sql/queries/from-using-pivot-and-unpivot.md]:
```suggestion
The following syntax summarizes how to use the `PIVOT` and `UNPIVOT` operator.  
```

**Comment by rwestMSFT** [comment]
@Rob-S Thank you for your contributions. We're going to create an internal PR from this, to allow us freshness pass over the existing content, and include your changes as discussed with @VanMSFT.

On a personal note, I want to also thank you for another contribution you made some time ago regarding `QUOTED_IDENTIFIER`, which was merged last week.

**Comment by Rob-S** [comment]
@rwestMSFT Just in summary for @VanMSFT, it originally looked like the optional clauses were not being highlighted with "[ ... ]" (as evidenced by the word "optional" in the "optional ORDER BY clause" part). Hence my original commit was simply to surround the column name placeholders properly with "< ... >". That's the simplest fix. As we later got into the discussion about optional clauses, that's what resulted in the more elaborate meta-syntax to surround the full-length (ex: including the AS <alias> part) optional clauses with "[ ... ]". Thanks!

**Pull request closed without merging**